### PR TITLE
[core][autoscaler] make logging level configurable

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

There is no way to configure logging level for KubeRay Autoscaler. This PR makes it configurable.

## Checks

1. Deploy a RayCluster using https://gist.github.com/kevin85421/3d48b2c36d46b826cf4299677d8d75d8, which sets the autoscaler log level to `DEBUG`.

2. Check Autoscaler logs
  image

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by dentiny — APPROVED** [review]

**Comment by rynewang** [comment]
this means LOGGER_LEVEL no longer applies to autoscaler?

**Comment by kevin85421** [comment]
> this means LOGGER_LEVEL no longer applies to autoscaler?

Yes, `ray_constants.LOGGER_LEVEL` is hardcoded to "info," and the environment variable also defaults to "INFO".


https://github.com/ray-project/ray/blob/0a11215e878025bc9810ebb7cc33f360d7251491/python/ray/_private/ray_constants.py#L224

**Comment by MengjinYan** [comment]
> > this means LOGGER_LEVEL no longer applies to autoscaler?
> 
> Yes, `ray_constants.LOGGER_LEVEL` is hardcoded to "info," and the environment variable also defaults to "INFO".
> 
> https://github.com/ray-project/ray/blob/0a11215e878025bc9810ebb7cc33f360d7251491/python/ray/_private/ray_constants.py#L224

Just curious, is `LOGGER_LEVEL` a universal default for all loggers? If that's the case, wondering if the default autoscalar log level should be set to `LOGGER_LEVEL` instead of the hardcoded value?

File: python/ray/autoscaler/_private/kuberay/run_autoscaler.py [hunk 1]
```diff
@@ -96,9 +96,11 @@ def _setup_logging() -> None:
     # The director should already exist, but try (safely) to create it just in case.
     try_to_create_directory(log_dir)
 
+    logging_level = ray_constants.AUTOSCALER_LOG_LEVEL
+
     # Write logs at info level to monitor.log.
     setup_component_logger(
-        logging_level=ray_constants.LOGGER_LEVEL,  # info
```
Comment (rickyyx) [inline python/ray/autoscaler/_private/kuberay/run_autoscaler.py:101]: I wonder if we should make this ray_constants.LOGGER_LEVEL a configurable thing through env actually.

**Review by rickyyx — APPROVED** [review]

**Comment by kevin85421** [comment]
> Just curious, is LOGGER_LEVEL a universal default for all loggers?

No, the logging level configurations are quite fragmented in the Ray codebase. Some default to `LOGGER_LEVEL`, some are hardcoded, and others default to different variables.

> I wonder if we should make this ray_constants.LOGGER_LEVEL a configurable thing through env actually.

I'm not sure whether we should have this kind of coarse-grained configuration or not. The logging level configurations are quite fragmented in the Ray codebase, making it difficult for users to understand the expected behavior of setting `LOGGER_LEVEL` (which loggers will be affected and which will not).

**Comment by rickyyx** [comment]
> The logging level configurations are quite fragmented in the Ray codebase

Is this a desired behaviour? i am curious of what benefits a fine-grained logging configuration would actually bring.

**Comment by kevin85421** [comment]
> Is this a desired behaviour?

I don't think so. 

> i am curious of what benefits a fine-grained logging configuration would actually bring.

The only benefit I can think of is that it makes it easier for users to understand the expected behavior, so they are less likely to mistake a behavior for a bug.

**Comment by kevin85421** [comment]
I updated the PR to make `LOGGER_LEVEL` configurable. For now, I don't encourage Ray users to configure it until its scope and expected behavior are clear and easy to understand. Currently, only Ray developers should use it.

**Comment by kevin85421** [comment]
I updated the PR description and how do I test it.

**Pull request merged**